### PR TITLE
Support Multiple Instance Types

### PR DESCRIPTION
**Pull request by mcaulifn**
This PR adds support for multiple spot instance types.

- Creates a launch template for each type
- Set the default template version to `$LATEST`

Changes from #871 merged to this branch. That PR needs to be merged first.

### CODE REVIEW

**Comment by mcaulifn** [comment]
@npalm should be good to go on this one

**Comment by mcaulifn** [comment]
Fixes #569

**Comment by npalm** [comment]
@mcaulifn did you find anyway to test the fallback loop?

**Comment by mcaulifn** [comment]
> @mcaulifn did you find anyway to test the fallback loop?

Do you mean in unit tests or deployed?

**Comment by npalm** [comment]
I mean deployed

File: modules/runners/variables.tf [hunk 1]
```diff
@@ -63,6 +63,12 @@ variable "instance_type" {
   default     = "m5.large"
 }
 
+variable "instance_types" {
+  description = "List of instance types for the action runner."
```
Comment (npalm) [inline modules/runners/variables.tf:67]: Maybe we could move the default instance type from `instance_type` to the default for `instances_types` and mark the variable `instance_type` as depcrecated.

File: variables.tf [hunk 2]
```diff
@@ -354,3 +354,9 @@ variable "volume_size" {
   type        = number
   default     = 30
 }
+
+variable "instance_types" {
+  description = "List of instance types for the action runner."
```
Comment (npalm) [inline variables.tf:359]: See comment for `instances_types` in the module

**Review by npalm — COMMENTED** [review]
@mcaulifn thanks for this great contribution! Code changes looks good to me. Add once small suggestion. Will check today a deployment. Once done will update the PR.

**Comment by npalm** [comment]
Pull request verified, all working. Fall back is not checked in a actual deployment. @mcaulifn please can you check my minor review comment?

File: modules/runners/variables.tf (hunk 1 above)
Comment (mcaulifn) [reply modules/runners/variables.tf:67]: Works for me!

**Comment by mcaulifn** [comment]
I am working on building and deploying the lambdas today.

**Comment by mcaulifn** [comment]
Lambdas re-deployed and working as expected. I cannot simulate a spot capacity failure so that part is difficult to test.

**Review by npalm — APPROVED** [review]

**Comment by npalm** [comment]
@mcaulifn thanks!

**Comment by npalm** [comment]
@mcaulifn FYI oday find out that the scaling down was borken. Not sure what exectly caused the issue. A new lib of auth-app or a change in a PR. Problem was also not catched by any unit test. Was able to provide a quick fix. See also https://github.com/philips-labs/terraform-aws-github-runner/pull/905 Thanks for your great help!

**Pull request merged**